purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -107,7 +108,9 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
     private val dispatcher: Dispatcher,
     private val identityManager: IdentityManager,
     private val subscriberAttributesManager: SubscriberAttributesManager,
-    @set:JvmSynthetic @get:JvmSynthetic internal var appConfig: AppConfig
+    @set:JvmSynthetic @get:JvmSynthetic internal var appConfig: AppConfig,
+    apiKeyValidator: APIKeyValidator,
+    apiKey: String
```

Comment:
Right, on a first iteration I had it there. However, the `configure` method is not currently tested in Android, which would leave a gap on our test to make sure that the validation actually happened. After discussing with Andy, he suggested moving the validation to the purchases constructor, which would allow for easier testing than on the `configure` method.

Having said that, I agree it's not ideal having to add 2 extra parameters only for this and we have already created a task to add tests for the `configure` method (https://revenuecats.atlassian.net/browse/CSDK-101). So we could potentially move this back there after those tests have been added. I'm working on this task today, so we can hold this to after that's done

Reply:
Did the refactor for the `configure` method in https://github.com/RevenueCat/purchases-android/pull/543 and made this PR point to that one. Lmk what you think!